quantumblacklabs/kedro
# [KED-1555] Expose VERSION_FORMAT constant in Kedro IO library

## Description
I’m currently trying to parse versions, and wanted to see if I could avoid hardcoding the version format myself.

This is my code:
```python3
from datetime import datetime
last_load_version = datetime.strptime(
    catalog.datasets.data_set_name.get_last_load_version(),  # Replace with `resolve_load_version` in future version of Kedro
    "%Y-%m-%dT%H.%M.%S.%fZ",
)
last_load_version
```

## Development notes
Manual test:
```python3
>>> from datetime import datetime
>>> 
>>> from kedro.io.core import VERSION_FORMAT
>>> 
>>> datetime.strptime(generate_timestamp(), VERSION_FORMAT)
datetime.datetime(2020, 3, 21, 21, 33, 59, 405000)
>>> 
```

~~I'll add a unit test verifying converted-back timestamp is correct (no microseconds) later; I'm currently having issues with local tests and CircleCI... =/ (all local test failures seemed completely unrelated)~~ Test added.

## Checklist

- [ ] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](../RELEASE.md) file

## Notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
